example/vectorAdd/src/main.cpp

```diff
@@ -73,21 +73,42 @@ class VectorAddKernel
     }
 };
 
+template<class A, class B>
+#if defined(ALPAKA_ACC_CPU_BT_OMP4_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuOmp4<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuOmp2Blocks<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuFibers<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuOmp2Threads<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuSerial<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuThreads<A,B>;
+#elif defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+using DefaultAcc = alpaka::acc::AccGpuCudaRt<A,B>;
+#else
+class Stub;
+#define NOP
+#warning "No supported backend selected."
+#endif
+
 auto main()
 -> int
 {
-// This example is hard-coded to use the sequential backend.
-#if defined(ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED)
+#ifndef NOP
 
     // Define the index domain
     using Dim = alpaka::dim::DimInt<1u>;
     using Idx = std::size_t;
 
     // Define the accelerator
-    using Acc = alpaka::acc::AccCpuSerial<Dim, Idx>;
+    using Acc = DefaultAcc<Dim, Idx>;
     using DevAcc = alpaka::dev::Dev<Acc>;
     using PltfAcc = alpaka::pltf::Pltf<DevAcc>;
     using QueueAcc = alpaka::queue::QueueCpuBlocking;
+    std::cout << "Using alpaka device: " << typeid(Acc).name() << std::endl;
```

Comment:
Could you please fix the text and write `accelerator` instead of `device`?

Reply:
Fixed.

Btw.: This line prints out the mangled name. Do you know a way to do some c++filt on it in-line? Strangle, the c++filt on my laptop does not demangle this name.

Reply:
`alpaka::acc::getAccName<Acc>()` could give you what you want.

Reply:
Besides from this, there is no easy platform independent way to get the demangled name.

Reply:
Added the getAccName()